frontend/src/metabase/collections/components/CollectionContent/CollectionItemsTable.tsx

```diff
@@ -0,0 +1,200 @@
+
+import {
+  ALL_MODELS,
+  COLLECTION_PAGE_SIZE,
+} from "metabase/collections/components/CollectionContent/constants";
+import CollectionEmptyState from "metabase/collections/components/CollectionEmptyState";
+import type {
+  CreateBookmark,
+  DeleteBookmark,
+} from "metabase/collections/types";
+import { isRootTrashCollection } from "metabase/collections/utils";
+import { ItemsTable } from "metabase/components/ItemsTable";
+import { PaginationControls } from "metabase/components/PaginationControls";
+import CS from "metabase/css/core/index.css";
+import Search from "metabase/entities/search";
+import { usePagination } from "metabase/hooks/use-pagination";
+import type Database from "metabase-lib/v1/metadata/Database";
+import type {
+  Bookmark,
+  Collection,
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+import { SortDirection, type SortingOptions } from "metabase-types/api/sorting";
+
+import {
+  CollectionEmptyContent,
+  CollectionTable,
+} from "./CollectionContent.styled";
+
+export type CollectionItemsTableProps = {
+  collectionId: CollectionId;
+} & Partial<{
+  bookmarks: Bookmark[];
+  clear: () => void;
+  collection: Collection;
+  createBookmark: CreateBookmark;
+  databases: Database[];
+  deleteBookmark: DeleteBookmark;
+  getIsSelected: (item: CollectionItem) => boolean;
+  handleCopy: (items: CollectionItem[]) => void;
+  handleMove: (items: CollectionItem[]) => void;
+  hasPinnedItems: boolean;
+  loadingPinnedItems: boolean;
+  models: CollectionItemModel[];
+  pageSize: number;
+  selectOnlyTheseItems: (items: CollectionItem[]) => void;
+  selected: CollectionItem[];
+  toggleItem: (item: CollectionItem) => void;
+  onClick: (item: CollectionItem) => void;
+  showActionMenu: boolean;
+}>;
+
+export const CollectionItemsTable = ({
+  collectionId,
+  collection,
+  getIsSelected,
+  selectOnlyTheseItems,
+  databases,
+  bookmarks,
+  createBookmark,
+  deleteBookmark,
+  loadingPinnedItems,
+  hasPinnedItems,
+  selected,
+  toggleItem,
+  clear,
+  handleMove,
+  handleCopy,
+  pageSize = COLLECTION_PAGE_SIZE,
+  models = ALL_MODELS,
+  onClick,
+  showActionMenu = true,
+}: CollectionItemsTableProps) => {
+  const [unpinnedItemsSorting, setUnpinnedItemsSorting] =
+    useState<SortingOptions>({
+      sort_column: "name",
+      sort_direction: SortDirection.Asc,
+    });
+
+  const [total, setTotal] = useState<number | null>(null);
+
+  const { handleNextPage, handlePreviousPage, setPage, page, resetPage } =
+    usePagination();
+
+  useEffect(() => {
+    if (collectionId) {
+      resetPage();
+      setTotal(null);
+    }
+  }, [collectionId, resetPage]);
+
+  const handleUnpinnedItemsSortingChange = useCallback(
+    (sortingOpts: SortingOptions) => {
+      setUnpinnedItemsSorting(sortingOpts);
+      setPage(0);
+    },
+    [setPage],
+  );
+
+  const showAllItems = !collection || isRootTrashCollection(collection);
```

Comment:
I know this works, but the API is hide too much implications here, I'd just love to have this renamed to `isSdk` or something or even use `getIsEmbeddingSdk` selector e.g.

```suggestion
  const showAllItems = isEmbeddingSdk || isRootTrashCollection(collection);
```

I think this should convey the intent a lot better.